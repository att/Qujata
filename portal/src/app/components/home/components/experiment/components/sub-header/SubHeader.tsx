import styles from './SubHeader.module.scss';
import { Link } from "react-router-dom";
import ArrowLeftSvg from '../../../../../../../../src/assets/images/arrow-left.svg';

interface SubHeaderProps {
    linkTitle: string;
}
export const SubHeader: React.FC<SubHeaderProps> = (props: SubHeaderProps) => {
    return (
        <div className={styles.sub_header_wrapper}>
          <Link className={styles.back_link} to="/">
            <img className={styles.arrow_icon} src={ArrowLeftSvg} alt="arrow" />
            {props.linkTitle}
          </Link>
        </div>
    );
}

