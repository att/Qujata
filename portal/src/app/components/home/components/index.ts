export * from './experiment';
export * from './latest-experiments';
