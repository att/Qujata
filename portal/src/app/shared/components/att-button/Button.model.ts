export enum ButtonActionType {
  BUTTON = 'button',
  SUBMIT = 'submit',
  RESET = 'reset',
}

export enum ButtonStyleType {
  PRIMARY,
  SECONDARY,
  TEXT,
  WRAPPER,
}

export enum ButtonSize {
  SMALL,
  MEDIUM,
  LARGE,
  NONE,
}

export interface IButton {
  size: ButtonSize;
  styleType: ButtonStyleType;
  actionType: ButtonActionType;
  className?: string;
  disabled?: boolean;
  text: string;
  ariaLabel?: string;
  form?: string;
  e2eId?: string;
  onClick: (e?: React.MouseEvent<HTMLButtonElement>) => void;
}
