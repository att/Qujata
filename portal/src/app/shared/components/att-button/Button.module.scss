@import './../../../../styles/variables-keys';
@import './Button.reference';

$txtColor: var($attCobalt);

.button {
  border: none;
  font-size: $buttonFontSize;
  outline: none;
  @extend %basicButtonStyle;

  &:not(:disabled) {
    cursor: pointer;
  }

  &:disabled {
    @extend %disableStyle;
  }
}

.wrapper_button, .txt_button {
  &:disabled {
    background-color: transparent;
    border: 0;
  }
}

.primary_button:not(:disabled) {
  @extend %primaryStyle;
}

.secondary_button:not(:disabled) {
  @extend %secondaryStyle;
}

.txt_button {
  @extend %buttonReset;
  font: inherit;
  block-size: auto;
  min-inline-size: auto;
  text-align: left;

  &:not(:disabled) {
    &:not(:hover) {
      color: $txtColor;
    }

    &:hover {
      color: var($actionTextHoverColor);
      text-decoration: underline;
    }
  }
}

.wrapper_button {
  @extend %buttonReset;
  min-inline-size: 0;
  padding: 0;
}

.sm_button {
  @extend %smallSize;
  min-inline-size: 80px;
}

.md_button {
  @extend %mediumSize;
}

.lg_button {
  @extend %largeSize;
}
