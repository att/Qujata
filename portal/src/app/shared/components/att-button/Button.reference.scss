@import 'src/styles/variables-keys';

$primaryTextColor: var($attCobalt);
$primaryBorderColor: var($attCobalt);
$primaryHoverBackgroundColor: var($attCobalt);

$secondaryTextColor: #c70032;
$secondaryBorderColor: #c70032;
$secondaryBackgroundColor: white;
$secondaryBorder: white;
$secondaryHoverBackgroundColor: #c70032;

$disableBackgroundColor: #eeeeee;
$disableBorderColor: #d8d8d8;
$disableTextColor: #B7BABF;

$buttonFontSize: 16px;
$smallButtonSize: 28px;
$mediumButtonSize: 36px;
$largeButtonSize: 48px;

%buttonReset {
  background-color: transparent;
  box-shadow: none;
  border: 0;
  outline: none;
  color: var($attCobalt);
}

%basicButtonStyle {
  font-family: var($fontRegular);
  min-inline-size: 80px;
  padding: 0 16px;
  white-space: nowrap;
}

%primaryStyle {
  border: 1px solid $primaryHoverBackgroundColor;
  border-radius: 2px;
  color: var($primaryWhite);
  background-color: $primaryHoverBackgroundColor;

}

%secondaryStyle {
  background-color: transparent;
  color: $secondaryTextColor;
  border: solid 1px $secondaryBorderColor;
  &:hover, &:focus  {
    background-color: $secondaryHoverBackgroundColor;
    border-color: $secondaryHoverBackgroundColor;
    color: var($primaryWhite);
  }
}

%disableStyle {
  background-color: $disableBackgroundColor;
  border: solid 1px $disableBorderColor;
  color: $disableTextColor;
  cursor: initial;
  pointer-events: none;

  i {
    color: $disableTextColor;
  }
}

%smallSize {
  block-size: $smallButtonSize;
  font-size: 13px;
}

%mediumSize {
  block-size: $mediumButtonSize;
  font-size: 14px;
}

%largeSize {
  block-size: $largeButtonSize;
  font-size: 16px;
}
