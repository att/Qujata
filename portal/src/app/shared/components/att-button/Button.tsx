import { useCallback } from 'react';
import cn from 'classnames';
import styles from './Button.module.scss';
import { ButtonActionType, ButtonSize, ButtonStyleType } from './Button.model';
import { ButtonCssClassBySize, ButtonCssClassByStyleType } from './Button.const';

export interface ButtonProps {
  actionType: ButtonActionType;
  ariaLabel?: string;
  className?: string;
  disabled?: boolean;
  size: ButtonSize;
  styleType: ButtonStyleType;
  onButtonClick: (e: React.MouseEvent<HTMLButtonElement>) => void;
  children: React.ReactNode;
  form?: string;
  title?: string;
  e2eId?: string;

}

export const Button: React.FC<ButtonProps> = (props: ButtonProps) => {
  const { onButtonClick } = props;
  const onClick: (e: React.MouseEvent<HTMLButtonElement>) => void = useCallback((e: React.MouseEvent<HTMLButtonElement>): void => {
    e.currentTarget.blur();
    onButtonClick(e);
  }, [onButtonClick]);

  const ButtonElement: JSX.Element = (
    <button
      className={cn(props.className, styles.button, ButtonCssClassByStyleType.get(props.styleType), ButtonCssClassBySize.get(props.size))}
      disabled={props.disabled}
      type={props.actionType}
      aria-label={props.ariaLabel}
      onClick={onClick}
      form={props.form}
      title={props.title}
      data-cy={props.e2eId}
    >
      {props.children}
    </button>
  );

  return ButtonElement;
};
