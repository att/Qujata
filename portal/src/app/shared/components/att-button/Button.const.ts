import styles from './Button.module.scss';
import { ButtonSize, ButtonStyleType } from './Button.model';

export const ButtonCssClassByStyleType: Map<ButtonStyleType, string> = new Map<ButtonStyleType, string>([
  [ButtonStyleType.PRIMARY, styles.primary_button],
  [ButtonStyleType.SECONDARY, styles.secondary_button],
  [ButtonStyleType.TEXT, styles.txt_button],
  [ButtonStyleType.WRAPPER, styles.wrapper_button],
]);

export const ButtonCssClassBySize: Map<ButtonSize, string> = new Map<ButtonSize, string>([
  [ButtonSize.SMALL, styles.sm_button],
  [ButtonSize.MEDIUM, styles.md_button],
  [ButtonSize.LARGE, styles.lg_button],
  [ButtonSize.NONE, ''],
]);
