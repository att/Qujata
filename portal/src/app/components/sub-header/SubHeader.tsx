import { Button, ButtonActionType, ButtonSize, ButtonStyleType } from '../../shared/components/att-button';
import styles from './SubHeader.module.scss';
import { SUB_HEADER_EN } from './translate/en';
import CloseSvg from '../../../assets/images/close.svg';
import ArrowDownSvg from '../../../assets/images/arrow-down-icon.svg';
import ArrowUpDownSvg from '../../../assets/images/arrow-up.svg';
import cn from 'classnames';
import { SubHeaderItems } from './constants/sub-header.const';
import { ISubHeaderItem } from './models/sub-header.interface';
import { useCallback, useState } from 'react';

const CloseAriaLabel: string = 'Close';
const ToggleAriaLabel: string = 'Toggle';

export interface SubHeaderProps {
    handleCloseClick: () => void;
}
export const SubHeader: React.FC<SubHeaderProps> = (props: SubHeaderProps) => {
  const [open, setOpen] = useState<boolean>(true);

  const handleToggleClick: () => void = useCallback((): void => {
    setOpen((prev: boolean) => !prev);
  }, []);

  return (
    <div className={cn(styles.sub_header_wrapper, { [styles.close]: !open })}>
        <div className={cn(styles.sub_header_action_buttons, { [styles.sub_header_action_buttons_close]: !open })}>
            <Button
                className={styles.toggle_button}
                ariaLabel={CloseAriaLabel}
                size={ButtonSize.NONE}
                styleType={ButtonStyleType.WRAPPER}
                actionType={ButtonActionType.BUTTON}
                onButtonClick={handleToggleClick}
            >
                <img className={styles.arrow_icon} src={open ? ArrowUpDownSvg : ArrowDownSvg} alt={ToggleAriaLabel} />
                {SUB_HEADER_EN.TOGGLE_BUTTON}
            </Button>
            {open && <Button
                className={styles.close_button}
                ariaLabel={CloseAriaLabel}
                size={ButtonSize.NONE}
                styleType={ButtonStyleType.WRAPPER}
                actionType={ButtonActionType.BUTTON}
                onButtonClick={props.handleCloseClick}
            >
                <img className={styles.close_icon} src={CloseSvg} alt={CloseAriaLabel} />
            </Button>}
        </div>
        <div className={cn({ [styles.items_wrapper_open]: open }, { [styles.items_wrapper_close]: !open }, styles.items_wrapper)}>
            {SubHeaderItems.map((item: ISubHeaderItem, index: number) => 
                <div className={styles.item} key={item.alt}>
                    <div className={styles.item_icon_wrapper}>
                        <div className={styles.item_icon}>
                            <img src={item.icon} alt={item.alt} />
                        </div>
                        <div className={cn({[styles.sperator]: (index !== SubHeaderItems.length - 1)})} />
                    </div>
                    <div className={styles.item_description}>{item.description}</div>
                </div>
            )}
        </div>
    </div>
  );
};
