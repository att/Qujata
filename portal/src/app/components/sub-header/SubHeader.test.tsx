import { render, RenderResult } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { Button } from '../../shared/components/att-button';
import { SubHeader } from './SubHeader';

jest.mock('../../shared/components/att-button');

describe('SubHeader', () => {
  test('renders sub header ', () => {
    (Button as jest.Mock).mockImplementation(() => <div>Button</div>);
    const { container, getByText }: RenderResult = render(<SubHeader handleCloseClick={jest.fn()} />, { wrapper: MemoryRouter });
    expect(container.firstChild).toMatchSnapshot();
    expect(getByText('Thatâ€™s what we are doing on each iteration:')).toBeTruthy();
  });
});