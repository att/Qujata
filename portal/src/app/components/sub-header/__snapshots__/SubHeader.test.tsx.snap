// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubHeader renders sub header  1`] = `
<div
  class="sub_header_wrapper"
>
  <div>
    Button
  </div>
  <div
    class="sub_header_title"
  >
    Thatâ€™s what we are doing on each iteration:
  </div>
  <div
    class="items_wrapper"
  >
    <div
      class="item"
    >
      <div
        class="item_icon_wrapper"
      >
        <div
          class="item_icon"
        >
          <img
            alt="Handshake"
            src="handshake.svg"
          />
        </div>
        <div
          class="sperator"
        />
      </div>
      <div
        class="item_description"
      >
        Initiating handshake
using the selected
cryptographic algorithm.
      </div>
    </div>
    <div
      class="item"
    >
      <div
        class="item_icon_wrapper"
      >
        <div
          class="item_icon"
        >
          <img
            alt="Keys"
            src="keys.svg"
          />
        </div>
        <div
          class="sperator"
        />
      </div>
      <div
        class="item_description"
      >
        Exchanging keys using
the selected algorithm to
create a shared secret.
      </div>
    </div>
    <div
      class="item"
    >
      <div
        class="item_icon_wrapper"
      >
        <div
          class="item_icon"
        >
          <img
            alt="Lock"
            src="lock.svg"
          />
        </div>
        <div
          class="sperator"
        />
      </div>
      <div
        class="item_description"
      >
        Deriving symmetric keys,
establishing a secure channel
for communication.
      </div>
    </div>
    <div
      class="item"
    >
      <div
        class="item_icon_wrapper"
      >
        <div
          class="item_icon"
        >
          <img
            alt="Download"
            src="download.svg"
          />
        </div>
        <div
          class="sperator"
        />
      </div>
      <div
        class="item_description"
      >
        Downloading content
over the secure channel.
      </div>
    </div>
    <div
      class="item"
    >
      <div
        class="item_icon_wrapper"
      >
        <div
          class="item_icon"
        >
          <img
            alt="Check"
            src="check.svg"
          />
        </div>
        <div
          class=""
        />
      </div>
      <div
        class="item_description"
      >
        Closing
the session
      </div>
    </div>
  </div>
</div>
`;
